modules/unsupported/geojson-core/src/main/java/org/geotools/data/geojson/PagingFeatureCollection.java

```diff
@@ -77,27 +87,30 @@ protected SimpleFeatureCollection readNext(ObjectNode next) throws IOException {
 
     protected class PagingFeatureIterator implements SimpleFeatureIterator {
         private SimpleFeatureIterator delegate;
-        private ObjectNode next;
+        private ObjectNode nextPage;
 
         public PagingFeatureIterator(SimpleFeatureIterator delegate, ObjectNode next) {
             this.delegate = delegate;
-            this.next = next;
+            this.nextPage = next;
         }
 
         @Override
         public boolean hasNext() {
             // jump to the next link, if possible, until a non-empty collection is found,
             // or no next links are found
-            while (!delegate.hasNext() && next != null) {
+            while (!delegate.hasNext() && nextPage != null) {
                 try {
                     // call the reader again, but do not delegate to avoid creating a series of
                     // nested objects (the next collection might contain another and so on)
-                    SimpleFeatureCollection features = readNext(next);
-                    this.delegate = features.features();
+                    SimpleFeatureCollection features = readNext(nextPage);
                     if (features instanceof PagingFeatureCollection) {
-                        this.next = ((PagingFeatureCollection) features).next;
+                        this.nextPage = ((PagingFeatureCollection) features).getNext();
+                        this.delegate = ((PagingFeatureCollection) features)
+                                .getFirstCollection()
+                                .features();
                     } else {
-                        this.next = null;
+                        this.nextPage = null;
+                        this.delegate = features.features();
```

Comment:
This is the actual fix I guess?

Reply:
Yes, this is the fix. The delegate should point at a ListFeatureCollection.features and not a PagingFeatureCollection.features. There could have been a clearer separation between this in the code.